File: server/src/summary/summary.model.go
package summary

import (
	"github.com/ZombieHippie/test-gin/server/src/repo"
	"github.com/jinzhu/gorm"
	"time"
)

// Summary is a common object for reports to point at
type Summary struct {
	gorm.Model
	Repository   repo.Repository `gorm:"ForeignKey:RepositoryID"`
	RepositoryID string
	BranchID     string
	BuildID      int
	Commit       string

	Message string
	Author  string
	Success bool
	Created time.Time
}

File: drone-plugin/lib-ts/summary/summary.model.ts

import { Repository } from "../repo/repo.model"
import { Artifact } from "../artifact/artifact.model"

interface Summary {
  ID?:          number // Assigned by gorm
  Repository:   Repository // Assign this yourself
  RepositoryID: string
  BranchID:     string // "feature/no-more-bugs"
  BuildID:      number // 2
  Commit:       string // Commit hash

  Author?:  string
  Message?: string
  Success:  boolean
  Created:  Date // Date Summary was recorded
}

export { Summary }
